### PR TITLE
fix(theme): form fields styling according to the guideline

### PR DESCRIPTION
**Pull request by ybaskaran**
**What is the problem this PR is trying to solve?**
Some of the form input fields styling doesn't follow UI guidelines http://guidelines.talend.com/document/92132#/ui-controls/input-field

**What is the chosen solution to this problem?**
Fixes styling for all the form input fields

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/theme/example/index.html [hunk 1]
```diff
@@ -1020,6 +1020,12 @@ <h1 id="forms">Forms</h1>
                 <div class="form-group has-error">
                   <input type="text" class="form-control" id="inputError">
                   <label class="control-label" for="inputError">Input error</label>
+                  <div>
+                    <p></p>
```
Comment (frassinier) [inline packages/theme/example/index.html]: Why is there an empty paragraph here?

File: packages/theme/example/index.html [hunk 2]
```diff
@@ -1054,6 +1060,7 @@ <h1 id="forms">Forms</h1>
 
                 <div class="form-group">
                   <select class="form-control" id="selectDefault">
+                    <option disabled default selected>Select an option</option>
```
Comment (frassinier) [inline packages/theme/example/index.html]: Dunno `default` attr, what's its purpose?

File: output/theme.sasslint.txt [hunk 3]
```diff
@@ -10,23 +10,33 @@ src/theme/_dropdowns.scss
...
-  280:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
-  293:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   44:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   44:25  error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   48:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   48:25  error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   52:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   60:10  error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   61:7   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+   73:11  error    A value of `0` is not allowed. `none` must be used       border-zero
```
Comment (frassinier) [inline output/theme.sasslint.txt]: Easy to fix

File: packages/theme/src/theme/_forms.scss [hunk 4]
```diff
@@ -338,6 +365,20 @@ form {
 			line-height: 1.5;
 		}
 	}
+
+	.form-group > input.form-control::placeholder {
+		color: $silver;
+	}
+
+	.form-control + label {
+		font-size: 14px;
```
Comment (frassinier) [inline packages/theme/src/theme/_forms.scss]: A var exists in bootstrap-theme with this value.

File: packages/theme/src/theme/_forms.scss [hunk 5]
```diff
@@ -301,8 +322,14 @@ form {
 
 		.error-detail {
 			padding-left: 0;
+			margin-top: -10px;
+			margin-bottom: 0;
 
 			.text-danger {
+				line-height: 15px;
+				font-size: $font-size-small;
+				font-weight: 400;
```
Comment (frassinier) [inline packages/theme/src/theme/_forms.scss]: Default value?

File: packages/theme/src/theme/_forms.scss [hunk 6]
```diff
@@ -301,8 +322,14 @@ form {
 
 		.error-detail {
 			padding-left: 0;
+			margin-top: -10px;
+			margin-bottom: 0;
 
 			.text-danger {
+				line-height: 15px;
```
Comment (frassinier) [inline packages/theme/src/theme/_forms.scss]: `line-height` was `14px` I guess (`line-height`) is there a real difference?

File: packages/theme/src/theme/_forms.scss [hunk 7]
```diff
@@ -130,6 +147,10 @@ form {
 		&[multiple] {
 			background: none;
 		}
+
+		option:default {
```
Comment (frassinier) [inline packages/theme/src/theme/_forms.scss]: Do you have any reference?

File: output/theme.sasslint.txt (hunk 3 above)
Comment (ybaskaran) [reply output/theme.sasslint.txt]: fixed

File: packages/theme/example/index.html (hunk 1 above)
Comment (ybaskaran) [reply packages/theme/example/index.html]: the error div with empty <p> is generated by react, the html is copied from other applications, PFA
error_div_with_empy_pg

File: packages/theme/example/index.html (hunk 2 above)
Comment (ybaskaran) [reply packages/theme/example/index.html]: To identify the placeholder to remove from the dropdown list when selecting a value, also we can't use disabled attribute as we may have real disabled values in the dropdown

File: packages/theme/src/theme/_forms.scss (hunk 7 above)
Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: sample reference included in the theme http://1094.talend.surge.sh/theme/#forms

File: packages/theme/src/theme/_forms.scss (hunk 6 above)
Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: makes a minor difference when the error message occupies more than one line, this was requested by Marc

File: packages/theme/src/theme/_forms.scss (hunk 5 above)
Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: yes, fixed

File: packages/theme/src/theme/_forms.scss (hunk 4 above)
Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: fixed

File: packages/theme/src/theme/_forms.scss (hunk 6 above)
Comment (talend-glorieux) [reply packages/theme/src/theme/_forms.scss]: OK. Could you make sure it's reflected on the guidelines?

Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: yes, it is already mentioned in the guidelines, Ref: "sizes" section under http://guidelines.talend.com/document/92132#/ui-controls/input-field

File: packages/theme/example/index.html (hunk 1 above)
Comment (jsomsanith) [reply packages/theme/example/index.html]: But here it's a demo page, it's an issue in the generated form, you can remove it

Comment (ybaskaran) [reply packages/theme/example/index.html]: removed the error message with empty paragraph

File: packages/theme/src/theme/_forms.scss [hunk 8]
```diff
@@ -333,9 +359,36 @@ form {
 	}
 
 	.form-group-lg {
-		select,
-		select.form-control {
-			line-height: 1.5;
+		select {
+			&.form-control {
```
Comment (jsomsanith) [inline packages/theme/src/theme/_forms.scss]: `.form-group-lg { select { &.form-control {} } }`--> `.form-group-lg select.form-control {}`

File: packages/theme/src/theme/_forms.scss [hunk 9]
```diff
@@ -333,9 +359,36 @@ form {
...
+		color: $silver;
+	}
+
+	.form-control + label {
+		font-size: $font-size-base;
+		color: $silver-chalice;
+	}
+
+	textarea {
+		&.form-control {
```
Comment (jsomsanith) [inline packages/theme/src/theme/_forms.scss]: `textarea { &.form-control {} }` --> `textarea.form-control {}`

File: packages/theme/src/theme/_forms.scss [hunk 10]
```diff
@@ -349,6 +402,27 @@ form {
 		select {
 			padding: 0;
 		}
+
+		&.multi-select-readonly {
```
Comment (jsomsanith) [inline packages/theme/src/theme/_forms.scss]: What is this class ?

Comment (frassinier) [reply packages/theme/src/theme/_forms.scss]: This it's not related to Bootstrap indeed

**Comment by frassinier** [comment]
Some stories look weird like [this one](http://1094.talend.surge.sh/forms/?knob-Arrays=%7B%22jsonSchema%22%3A%7B%22type%22%3A%22object%22%2C%22properties%22%3A%7B%22listOfStrings%22%3A%7B%22type%22%3A%22array%22%2C%22title%22%3A%22A%20list%20of%20strings%22%2C%22items%22%3A%7B%22type%22%3A%22string%22%2C%22default%22%3A%22bazinga%22%7D%7D%2C%22multipleChoicesList%22%3A%7B%22type%22%3A%22array%22%2C%22title%22%3A%22A%20multiple%20choices%20list%22%2C%22items%22%3A%7B%22type%22%3A%22string%22%2C%22enum%22%3A%5B%22foo%22%2C%22bar%22%2C%22fuzz%22%2C%22qux%22%5D%7D%2C%22uniqueItems%22%3Atrue%7D%2C%22fixedItemsList%22%3A%7B%22type%22%3A%22array%22%2C%22title%22%3A%22A%20list%20of%20fixed%20items%22%2C%22items%22%3A%5B%7B%22title%22%3A%22A%20string%20value%22%2C%22type%22%3A%22string%22%2C%22default%22%3A%22lorem%20ipsum%22%7D%2C%7B%22title%22%3A%22a%20boolean%20value%22%2C%22type%22%3A%22boolean%22%7D%5D%2C%22additionalItems%22%3A%7B%22title%22%3A%22Additional%20item%22%2C%22type%22%3A%22number%22%7D%7D%2C%22nestedList%22%3A%7B%22type%22%3A%22array%22%2C%22title%22%3A%22Nested%20list%22%2C%22items%22%3A%7B%22type%22%3A%22array%22%2C%22title%22%3A%22Inner%20list%22%2C%22items%22%3A%7B%22type%22%3A%22string%22%2C%22default%22%3A%22lorem%20ipsum%22%7D%7D%7D%7D%7D%2C%22uiSchema%22%3A%7B%22multipleChoicesList%22%3A%7B%22ui%3Awidget%22%3A%22checkboxes%22%7D%2C%22fixedItemsList%22%3A%7B%22items%22%3A%5B%7B%22ui%3Awidget%22%3A%22textarea%22%7D%2C%7B%22ui%3Awidget%22%3A%22select%22%7D%5D%2C%22additionalItems%22%3A%7B%22ui%3Awidget%22%3A%22updown%22%7D%7D%7D%2C%22properties%22%3A%7B%22listOfStrings%22%3A%5B%22foo%22%2C%22bar%22%5D%2C%22multipleChoicesList%22%3A%5B%22foo%22%2C%22bar%22%5D%2C%22fixedItemsList%22%3A%5B%22Some%20text%22%2Ctrue%2C123%5D%2C%22nestedList%22%3A%5B%5B%22lorem%22%2C%22ipsum%22%5D%2C%5B%22dolor%22%5D%5D%7D%7D&knob-KeyValueArray=%7B%22jsonSchema%22%3A%7B%22title%22%3A%22Key%2Fvalue%20form%22%2C%22description%22%3A%22A%20simple%20form%20example.%22%2C%22type%22%3A%22object%22%2C%22properties%22%3A%7B%22keyValue%22%3A%7B%22type%22%3A%22array%22%2C%22title%22%3A%22Key%2FValue%22%2C%22items%22%3A%7B%22type%22%3A%22string%22%7D%7D%7D%7D%2C%22uiSchema%22%3A%7B%22keyValue%22%3A%7B%22items%22%3A%7B%22ui%3Awidget%22%3A%22keyValue%22%7D%7D%7D%2C%22properties%22%3A%7B%22keyValue%22%3A%5B%22a%3Db%22%2C%22c%3Dd%22%5D%7D%7D&selectedKind=Migration&selectedStory=KeyValueArray&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

File: packages/theme/src/theme/_forms.scss (hunk 9 above)
Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: tag.className (select.form-control) fails with lint:style checks
"error    Qualifying elements are not allowed for class selectors  no-qualifying-elements"
so we are using "tag {  &.className { .......... } }"

File: packages/theme/src/theme/_forms.scss (hunk 8 above)
Comment (ybaskaran) [reply packages/theme/src/theme/_forms.scss]: tag.className (select.form-control) fails with lint:style checks
"error    Qualifying elements are not allowed for class selectors  no-qualifying-elements"
so we are using "tag {  &.className { .......... } }"

**Comment by ybaskaran** [comment]
Fixed review comments requested by @frassinier

File: packages/theme/example/index.html (hunk 2 above)
Comment (ybaskaran) [reply packages/theme/example/index.html]: removed the default attribute

**Pull request merged**